lib/daemon.js

```diff
@@ -68,21 +49,16 @@ class Daemon {
     Object.keys(this._pollers).forEach(pollerId => this._removePoller(pollerId))
   }
 
-  _addPoller (descriptor, forcePoll = false) {
-    this._logger.info('spinning up poller', descriptor)
-
-    const poller = new Poller({
-      backends: this._backends,
-      intervalMilliseconds: this._pollerIntervalMilliseconds,
-      kubeClient: this._kubeClient,
-      logger: this._logger,
-      metrics: this._metrics,
-      namespace: descriptor.namespace,
-      secretDescriptor: descriptor.secretDescriptor,
-      ownerReference: descriptor.ownerReference
-    })
-
-    this._pollers[descriptor.id] = poller.start({ forcePoll })
+  _addPoller (descriptor) {
+    this._logger.info('spinning up poller for', descriptor.name, 'in', descriptor.namespace)
+
+    if (this._pollers[descriptor.id]) {
+      throw new Error(`Poller for ${descriptor.id} already exists`)
+    }
```

Comment:
So after running this for a few hours, this seems to happen occasionally, not really sure whats triggering it. 

Whenever an external secret is polled we will update the status for this external secret, this in turn should trigger the watch stream with a modified event. Which means we recreate the poller and schedule the next poll. As we track the generation with observedGeneration we won't end up in an endless loop as generation is not updated by status updates.

Reply:
With change in bb3787c this now seems fine.

Switched back to returning in _upsertSecret here: https://github.com/godaddy/kubernetes-external-secrets/pull/133/commits/cf17824cde407c9bfed571b4bd72c20b1e9a3fe9
But this seemed to cause issues with errors getting caught in the wrong try catch. ie upserts failing. so added the await again https://github.com/godaddy/kubernetes-external-secrets/pull/133/commits/c8be7f389390a9e7de8046b2d28607f39c533a5a

Running test in our dev cluster again 🎉